main.go

```diff
@@ -304,7 +309,16 @@ func backendMain() error {
 						return
 					case payments := <-client.ReceivedPayments():
 						for _, payment := range payments {
-							doDeposit(engine, payment, asset)
+							if isExchangePayment(payment) {
+								// if we received exchange payment we need to
+								// notify about it exchange
+								doDeposit(engine, payment, asset)
+							} else {
+								// else we need to add it in payment store
```

Comment:
Adding to my comment above, probably we should just use another service to fetch payments from connector, and notify exchange about it, so that connector might be more generic, and wouldn't have dependencies with viabtc client libraries.

Reply:
Sure. We need some persistent database with notification support like postgres. But simple http-hook notifications maybe also useful.